prow/cmd/deck/main.go

```diff
@@ -948,7 +949,9 @@ func renderSpyglass(ctx context.Context, sg *spyglass.Spyglass, cfg config.Gette
 		return "", fmt.Errorf("error when resolving real path %s: %v", src, err)
 	}
 	src = realPath
-
+	if err := ValidateStoragePath(cfg, src); err != nil {
+		return "", err // TODO(eblackwelder@): indicate it's http.StatusBadRequest (from inside ValidateStoragePath)
```

Comment:
I have an idea how to address this, but would rather implement it in a separate patch:
* create an `httpError` type that passes a status code along with the error message (unless we already have that type)
* create a helper method to "getHttpStatusCode" for a given `error` (by default 500, but checking for httpError's status code)
* return an `httpError` in `ValidateStoragePath`
* use the `getHttpStatusCode` method in various places: `handleArtifactView`, etc

Reply:
That sounds reasonable. Alternatively `renderSpyglass` could just be updated to add an HTTP status code return value.
I think this should be handled in this patch though rather than add new code that we know to be incorrect.

Reply:
Implemented.